wsmaster/codenvy-hosted-api-system-license/src/main/java/com/codenvy/api/license/SystemLicenseFilter.java

```diff
@@ -0,0 +1,151 @@
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.UnauthorizedException;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.dto.server.DtoFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import static com.codenvy.api.license.shared.model.Issue.Status.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED;
+
+/**
+ * Checks system license conditions.
+ * If ("no.user.interaction" property == false) and (user has permission to perform MANAGE_SYSTEM_ACTION), then send redirection to accept-fair-source-license page in response.
+ * If ("no.user.interaction" property == true), then don't send redirection despite of whoever user is.
+ * @author Dmytro Nochevov
+ */
+@Singleton
+public class SystemLicenseFilter implements Filter {
+    public static final String NO_USER_INTERACTION                 = "no.user.interaction";
+    public static final String ACCEPT_FAIR_SOURCE_LICENSE_PAGE_URL = "license.system.accept_fair_source_license_page_url";
+
+    @Inject
+    protected RequestFilter          requestFilter;
+    @Inject
+    protected HttpJsonRequestFactory requestFactory;
+    @Inject
+    @Named("che.api")
+    protected String                 apiEndpoint;
+    @Inject
+    @Named(NO_USER_INTERACTION)
+    protected boolean                noUserInteraction;
+    @Inject
+    @Named(ACCEPT_FAIR_SOURCE_LICENSE_PAGE_URL)
+    protected String                 acceptFairSourceLicensePageUrl;
+
+    @Override
+    public void init(FilterConfig config) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        final HttpServletRequest httpRequest = (HttpServletRequest)request;
+        final HttpServletResponse httpResponse = (HttpServletResponse)response;
+        if (shouldSkip(httpRequest)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        try {
+            if (isFairSourceLicenseNotAccepted()) {
+                if (!noUserInteraction && isAdmin()) {
+                    sendUserToAcceptFairSourceLicensePage(httpResponse);
+                } else {
+                    createForbiddenAccessResponse(httpResponse);
```

Comment:
1. We should redirect to error page to see error in Login page itself:
![error_of_creation_user_using_create_account_page_when_fair_source_license_is_not_accepted](https://cloud.githubusercontent.com/assets/1197777/20934145/cef97e0a-bbe2-11e6-8c19-8e64e7a527f8.png)

2. You are right. We set `no.user.interaction = false` for OnPremisesIdeApiModule.